### DISCUSSION TITLE
Using Dialog for modals causes sticky elements to flicker

### DISCUSSION THREAD

**Discussion by EricDotSmith** [tailwindlabs/headlessui#2305] [Help] [answered]
I have a sticky element at the top of my page, when you open a modal with headlessui's Dialog tag, it causes the sticky div to flicker; meanwhile if you open the modal without headlessui's Dialog tag, the sticky div doesn't flicker. Why is this happening? Is there a way around this? I want to use the Dialog tag as it handles most of what I want from a modal, but I'm also not liking the flickering its introducing. This is happening on iOS on Safari.

Here is the repo used to demonstrate this issue: https://github.com/EricDotSmith/HeadlessUIDialogFlickerSafariiOS

**UPDATE:** _To add more context, it's clear that the overflow: hidden that's being applied to the html tag from somewhere within Dialog is causing this issue. Unfortunately, I still haven't figured out a way to remediate this issue._

https://user-images.githubusercontent.com/67427202/221332205-cbd0afee-5a8f-45a3-a952-b7c206f56edb.MP4

**Comment by EricDotSmith** [comment]
Related: https://github.com/tailwindlabs/headlessui/discussions/2181

**Comment by alex-carter-mobdev** [comment, marked as answer]
best solution 
html:has(dialog[open]) {
  /* remove the main scrollbar when dialog is open */
  overflow: hidden;
}
